orchestration/api/services.py

```diff
@@ -14,15 +14,114 @@
+
+    service_defs = []
+    for service_def in service_def_list:
+        wfs = []
+        sd_hash = {}
+        for sd, wd in service_workflow_list:
+            wfd_hash = {'id': wd.id,
+                        'name': wd.name,
+                        'description': wd.description,
+                        'definition': json.loads(wd.definition)
+                        }
+            if service_def['id'] == sd.id:
+                wfs.append(wfd_hash)
+        service_def['workflows'] = wfs
+        sd_hash['service'] = service_def
+        service_defs.append(sd_hash)
+
+    service_defs_hash = {'services': service_defs}
+
+    return jsonify(response=service_defs_hash), 200
+
+
+# This API will provide the details of the service 'id' provided
+@service.route(
+    "/v1beta/orchestration/services/<string:service_id>",
+    methods=['GET'])
+def get_services(service_id):
+    service_def_hash = get_service_def(service_id)
+    return jsonify(response=service_def_hash), 200
+
+
+# API to register the ServiceDefinitions to Orchestration Manager
+@service.route(
+    "/v1beta/orchestration/services",
+    methods=['POST'])
+def add_services():
+    payload = request.get_json()
+    service_data = json.loads(json.dumps(payload))
+
+    workflow_defs = get_wfds()
+    wf_hash = {}
+    wfs = []
+    wf_objs = []
+    for elem in workflow_defs.values():
+        if elem['runner_type'] == 'mistral-v2':
+            wfd_hash = {'id': elem['ref'],
+                        'name': elem['name'],
+                        'description': elem['description'],
+                        'definition': json.dumps(elem['parameters'])
+                        }
+            # check if the entries are not present in DB then only
+            # enter in DB
+            if get_workflow_definition(None, elem['ref']) is None and \
+                    re.search(service_data["name"],
+                              elem['ref'], re.IGNORECASE):
+                wf_obj = create_workflow_definition(None, wfd_hash)
+                wfs.append(wfd_hash)
+                wf_objs.append(wf_obj)
+            wf_hash['workflows'] = wfs
+
+    service_data['id'] = str(uuid.uuid4()).replace('-', '')
+
+    for wf_ob in wf_objs:
+        service_data["workflow_definitions"] = [wf_ob]
+        create_service_definition(None, service_data)
+
+    service_def_hash = get_service_def(service_data['id'])
+
+    return jsonify(response=service_def_hash), 200
+
+
+def get_service_def(service_id):
+    service_def_list = get_service_definition(service_id, None)
+    service_workflow_list = get_sd_wfd_association(service_id, None)
+
+    wfs = []
+    for sd, wd in service_workflow_list:
+        wfd_hash = {'id': wd.id,
+                    'name': wd.name,
+                    'description': wd.description,
+                    'definition': json.loads(wd.definition)
+                    }
+        wfs.append(wfd_hash)
+
+    sd_hash = service_def_list[0]
```

Comment:
if query returns query.first() then we may not need this i.e. getting first elem, right?

Reply:
We need to get the single(first) elem from the list.